Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,210 @@
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Windows.System.RemoteSystems;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Picker Control to show List of Remote Devices that are accessible
+    /// </summary>
+    public sealed class RemoteDevicePicker : ContentDialog
+    {
+        private Dictionary<string, RemoteSystem> DeviceMap { get; set; }
+
+        private ListView _listDevices;
+        private ComboBox _listDeviceTypes;
+        private ProgressRing _progressRing;
+
+        /// <summary>
+        /// Gets List of All Remote Systems based on Selection Filter
+        /// </summary>
+        public ObservableCollection<RemoteSystem> RemoteSystems { get; private set; }
+
+        /// <summary>
+        /// Gets or sets the Line Color on control Header. takes **SystemControlBackgroundAccentBrush** by default
+        /// </summary>
+        public Brush HeaderLineColor
+        {
+            get { return (Brush)GetValue(HeaderLineColorProperty); }
+            set { SetValue(HeaderLineColorProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="HeaderLineColor"/>.
+        /// </summary>
+        public static readonly DependencyProperty HeaderLineColorProperty = DependencyProperty.Register(
+            nameof(HeaderLineColor),
+            typeof(Brush),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(Application.Current.Resources["SystemControlBackgroundAccentBrush"]));
+
+        /// <summary>
+        /// Gets or sets the DeviceList Selection Mode. Defaults to ListViewSelectionMode.Single
+        /// </summary>
+        public ListViewSelectionMode DeviceListSelectionMode
+        {
+            get { return (ListViewSelectionMode)GetValue(DeviceListSelectionModeProperty); }
+            set { SetValue(DeviceListSelectionModeProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets the dependency property for <see cref="DeviceListSelectionMode"/>.
+        /// </summary>
+        public static readonly DependencyProperty DeviceListSelectionModeProperty = DependencyProperty.Register(
+            nameof(DeviceListSelectionMode),
+            typeof(ListViewSelectionMode),
+            typeof(RemoteDevicePicker),
+            new PropertyMetadata(ListViewSelectionMode.Single));
+
+        /// <summary>
+        /// Fired when the Remote Device Picker is Closed.
+        /// </summary>
+        public event EventHandler<RemoteDevicePickerEventArgs> RemoteDevicePickerClosed;
```

Comment:
With previous statement to have an async method to return the result this should be removed along with the event args class